### PR TITLE
Intermediate cache to volumes

### PR DESCRIPTION
**Pull request by ganeshdogiparthi-db**

### CODE REVIEW

File: src/databricks/labs/remorph/reconcile/compare.py [hunk 1]
```diff
@@ -21,26 +22,53 @@ def raise_column_mismatch_exception(msg: str, source_missing: list[str], target_
...
-        .drop(_HASH_COLUMN_NAME)
+        df.filter(col(f"{source_alias}_{_HASH_COLUMN_NAME}").isNull())
+        .select(
+            *[
+                col(col_name).alias(col_name.replace(f'{target_alias}_', '').lower())
+                for col_name in df.columns
+                if col_name.startswith(f'{target_alias}_')
+            ]
+        )
+        .drop(f"{_HASH_COLUMN_NAME}")
```
Comment (nfx) [inline src/databricks/labs/remorph/reconcile/compare.py:59]:
```suggestion
        .drop(_HASH_COLUMN_NAME)
```
F-string is redundant here

**Review by nfx — CHANGES_REQUESTED** [review]

File: src/databricks/labs/remorph/reconcile/compare.py (hunk 1 above)
Comment (ganeshdogiparthi-db) [reply src/databricks/labs/remorph/reconcile/compare.py]: Done

File: src/databricks/labs/remorph/reconcile/recon_capture.py [hunk 2]
```diff
@@ -28,6 +33,43 @@
...
+def clean_unmatched_df_from_volume(workspace_client: WorkspaceClient, path: str):
+    try:
+        workspace_client.dbfs.delete(path, recursive=True)
+    except Exception as e:
+        message = f"Error cleaning up unmatched DF from volumes --> {e}"
+        logger.error(message)
+        raise CleanFromVolumeException(message) from e
+
+
+def write_and_read_unmatched_df_with_volumes(
```
Comment (sundarshankar89) [inline src/databricks/labs/remorph/reconcile/recon_capture.py:59]:
```suggestion
def read_and_write_unmatched_df_with_volumes(
```

**Review by sundarshankar89 — CHANGES_REQUESTED** [review]

**Review by sundarshankar89 — APPROVED** [review]
LGTM

**Review by nfx — APPROVED** [review]

**Pull request merged**